From e05f05e7cfe9d1e3f34daaa824e3c9348691dd81 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 30 Sep 2016 15:19:06 -0400
Subject: [PATCH] Mogrify global buffer could overflow

Test is length is negative and bail out early

This fix CVE-2016-7799

bug: https://github.com/ImageMagick/ImageMagick/issues/280
bug-debian: https://bugs.debian.org/840437
origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/a7bb158b7bedd1449a34432feb3a67c8f1873bfa
---
 magick/profile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: imagemagick-6.6.9.7/magick/profile.c
===================================================================
--- imagemagick-6.6.9.7.orig/magick/profile.c	2016-11-29 13:46:49.009277539 -0500
+++ imagemagick-6.6.9.7/magick/profile.c	2016-11-29 13:46:49.005277493 -0500
@@ -1961,7 +1961,7 @@
       (void) AddValueToSplayTree(exif_resources,q,q);
       tag_value=(ssize_t) ReadProfileShort(endian,q);
       format=(ssize_t) ReadProfileShort(endian,q+2);
-      if ((format-1) >= EXIF_NUM_FORMATS)
+      if ((format < 0) || ((format-1) >= EXIF_NUM_FORMATS))
         break;
       components=(ssize_t) ReadProfileLong(endian,q+4);
       if (components < 0)
